src/ResourceManager/IotHub/Commands.IotHub/IotHub/SetAzureRmIotHub.cs

```diff
@@ -0,0 +1,248 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using System.Linq;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+
+    [Cmdlet(VerbsCommon.Set, "AzureRmIotHub", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSIotHub))]
+    public class SetAzureRmIotHub : IotHubBaseCmdlet
+    {
+        const string UpdateSkuParameterSet = "UpdateSku";
+        const string UpdateEventHubEndpointPropertiesParameterSet = "UpdateEventHubEndpointProperties";
+        const string UpdateFileUploadPropertiesParameterSet = "UpdateFileUploadProperties";
+        const string UpdateCloudToDevicePropertiesParameterSet = "UpdateCloudToDeviceProperties";
+        const string UpdateOperationsMonitoringPropertiesParameterSet = "UpdateOperationsMonitoringProperties";
+
+        [Parameter(
```

Comment:
If this is mandatory in all parameter sets, you can use a single parameter attribute with no ParameterSetName setting